sphinx/config.py

```diff
@@ -227,6 +227,7 @@ class Config:
         'template_bridge': _Opt(None, 'html', frozenset((str,))),
         'keep_warnings': _Opt(False, 'env', ()),
         'suppress_warnings': _Opt([], 'env', ()),
+        'show_warning_types': _Opt(False, 'env', ()),
```

Comment:
```suggestion
        'show_warning_types': _Opt(False, 'env', frozenset((bool,))),
```

I think we can force new configuration values to be of certain type.